### PR TITLE
feat: add IAM resources for Vertex AI Endpoints

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/14630

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
vertexai: Added IAM resources for google_vertex_ai_endpoint
```

### CODE REVIEW

**Comment by browningjp-datatonic** [comment]
hi @slevenick! I have now run the unit tests for `vertexai` and `make lint` successfully. Not sure what acceptance tests to run - are there specific ones to check for new IAM resources? Thanks

**Comment by browningjp-datatonic** [comment]
@slevenick would you be able to take a look at this?

**Comment by slevenick** [comment]
Sorry I must have missed this originally. Can you rebase this from current main? Our CI is having trouble running it as it's too old

**Comment by slevenick** [comment]
Tests seem not to be generating. Looks like it's because we skip the only test in the vertex AI endpoints resource.

Can we unmark the test as skip_test to see if that will generate the IAM resource tests?

**Comment by browningjp-datatonic** [comment]
@slevenick done!

**Comment by slevenick** [comment]
/gcbrun
Huh... I'm not super sure why this failed

I'll try kicking it off again

**Comment by slevenick** [comment]
Oh no, I see it: 

google/resource_vertex_ai_endpoint_test.go:150:6: testAccCheckVertexAIEndpointDestroyProducer redeclared in this block

We may need to go back to skipping the test and hand-writing a test for the IAM resources specifically. You can use the generated test in the previous diff as a starting place for that, because it should do all of what we need

**Comment by browningjp-datatonic** [comment]
@slevenick is this what you had in mind for the handwritten test?

**Comment by browningjp-datatonic** [comment]
@slevenick is anything more needed for this PR?

**Review by slevenick — CHANGES_REQUESTED** [review]
Sorry, there were some folder changes recently, you may need to rebase and move the handwritten file.

CONFLICT (file location): mmv1/third_party/terraform/tests/iam_vertex_endpoint_test.go.erb added in head/feat/vertex_endpoint_iam inside a directory that was renamed in HEAD, suggesting it should perhaps be moved to mmv1/third_party/terraform/services/compute/iam_vertex_endpoint_test.go.erb.

File: mmv1/third_party/terraform/services/vertexai/iam_vertex_endpoint_test.go.erb [hunk 1]
```diff
@@ -0,0 +1,407 @@
+<% autogen_exception -%>
+package google
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/vertexai/iam_vertex_endpoint_test.go.erb:2]: I think the package is different now to match vertexai

**Review by slevenick — CHANGES_REQUESTED** [review]

**Pull request closed without merging**